### DISCUSSION TITLE
Consistent renaming of "config" and "options".

### DISCUSSION THREAD

**Discussion by devanshj** [statelyai/xstate#2427] [Ideas]
According to [this comment and PR of David](https://github.com/statelyai/xstate/pull/1808#issuecomment-761590344), "config" has been renamed to "definition" and "options" have been renamed to "implementations". (Fun fact - txstate has those names since day one 😎 )

But he says "that would just be what's present in the documentation" which I think is overlooking the fact that `createMachine` returns an object that contains properties "config" and "options". So it would make sense that those are renamed too for consistency. Or else it would be awkward when the documentation says "You can get the implementations of the machine from the `options` property" xD

**Comment by davidkpiano** [comment]
That's a good idea; I'll make sure those are done for v5.

**Reply by davidkpiano** [reply]
Although `definition` may be tricky, since what's given to the machine isn't its definition, but a "partial" (sugary?) definition. The machine node itself has a `.definition` property which represents its full definition.

**Reply by devanshj** [reply]
In that case we'd have to name both differently. Or perhaps even omit the desugared version? Or keep only either of em? Not sure xD. If we're going with keeping both, these are few suggestions on top of my head -
1. `definition`, `desugaredDefinition`
2. `inputDefinition`, `definition`
3. `definition`, `concreteDefinition`
4. `design` (That's what [state-designer calls it I guess](https://www.state-designer.com/docs/api/createDesign)), `definition`

**Reply by geekox86** [reply]
If I may gentlemen, I prefer the singular "implementation" over the plural.

I also like "definition".

Sorry for intruding.
